modules/data.atmosphere/R/download.NEONmet.R

```diff
@@ -0,0 +1,398 @@
+##' @param outfolder location on disk where outputs will be stored
+##' @param start_date the start date of the data to be downloaded. Format is YYYY-MM-DD (will only use the year and month of the date)
+##' @param end_date the end date of the data to be downloaded. Format is YYYY-MM-DD (will only use the year and month part of the date)
+##' @param overwrite should existing files be overwritten
+##' @param verbose makes the function output more text
+##' @importFrom PEcAn.utils fqdn logger.debug logger.error logger.warn logger.severe
+
+download.NEONmet <- function(sitename, outfolder, start_date, end_date, 
+                                  overwrite = FALSE, verbose = FALSE,  ...) {
+
+  if (!file.exists(outfolder)) {
+    dir.create(outfolder)
+  }
+  
+  #Check if site is a NEON site
+  site <- sub(".* \\((.*)\\)", "\\1", sitename)
+  siteinfo <- nneo::nneo_site(site)
+  if (!exists("siteinfo")) {
+    logger.error("Could not get information about", sitename, ".", "Is this a NEON site?")
+  }
+  
+  #See what products and dates are available for this site
+  availProducts <- siteinfo$dataProducts$dataProductCode #list of data prodcuts by code
+  availDates <- siteinfo$dataProducts$availableMonths #lists of availably YYYY-MM per site, use unlist()
+  lat <- siteinfo$siteLatitude
+  lon <- siteinfo$siteLongitude
+  
+  #Figure out which months are needed
+  start_date <- as.POSIXlt(start_date, tz = "UTC")
+  end_date <- as.POSIXlt(end_date, tz = "UTC")
+  start_ymd <- lubridate::ymd(start_date)
+  end_ymd <- lubridate::ymd(end_date)
+
+  #Subset to max/min available dates if start or end date exceed those bounds
+  allDates <- unlist(availDates)
+  minDate <- min(allDates)
+  maxDate <- max(allDates)
+  start_ym <- substr(start_ymd,1,7)
+  end_ym <- substr(end_ymd,1,7)
+  if (start_ym<minDate) { 
+    start_ym <- minDate 
+  }
+  if (end_ym>maxDate) { 
+    end_ym <- maxDate 
+  }    
+  start_year <- as.numeric(substr(start_ym,1,4))
+  end_year <- as.numeric(substr(end_ym,1,4))
+  
+  #create results data frame
+  rows <- end_year - start_year + 1
+  results <- data.frame(file = character(rows), 
+                        host = character(rows), 
+                        mimetype = character(rows), 
+                        formatname = character(rows), 
+                        startdate = character(rows), 
+                        enddate = character(rows), 
+                        dbfile.name = paste0("NEONmet.",site), 
+                        stringsAsFactors = FALSE)
+  
+  all_years <- start_year:end_year
+  all_files <- file.path(outfolder, paste0("NEONmet.", site, ".", as.character(all_years), ".nc"))
+  results$file <- all_files
+  results$host <- fqdn()
+  results$mimetype   <- "application/x-netcdf"
+  results$formatname <- "CF"
+  results$startdate  <- paste0(all_years, "-01-01 00:00:00")
+  results$enddate    <- paste0(all_years, "-12-31 23:59:59")
```

Comment:
Assuming all of this has been tested, there's no need to change this. But for the future, it may be good to use functions specifically designed to process dates, either in base R, or functions from `lubridate`.